packages/salesforce/src/Adaptor.js

```diff
@@ -220,36 +223,54 @@ export function bulk(sObjectName, operation, records, options = {}) {
               reject(err);
             });
 
-            return batch
-              .on('queue', function (batchInfo) {
-                const batchId = batchInfo.id;
-                var batch = job.batch(batchId);
-                batch.poll(pollInterval, pollTimeout);
-              })
-              .then(async res => {
+            if (asyncMode) {
+              return batch.on('queue', async function (batchInfo) {
+                console.info('Batch queued. Batch ID:', batchInfo.id);
+                resolve({ batchInfo });
```

Comment:
A promise can only resolve once. If there are multiple queue events, for multiple chunks, then they'll never report back to user code. In other words, we'll only ever report back the first batch here.

Reply:
I have tested multiple batches by reducing the batch size to 1 and run integration test that insert two records with the option `asynMode: true`, I can see that i am getting two batches records after the bulk operation is done.